include/fiction/algorithms/simulation/sidb/calculate_energy_and_state_type.hpp

```diff
@@ -29,57 +30,54 @@ using sidb_energy_and_state_type = std::vector<std::pair<double, bool>>;
  * erroneous, transparent).
  *
  * @tparam Lyt SiDB cell-level layout type (representing a gate).
+ * @tparam TT The type of the truth table specifying the gate behavior.
  * @param energy_distribution Energy distribution.
- * @param output_cells SiDBs in the layout from which the output is read.
- * @param output_bits Truth table entry for a given input (e.g. 0 for AND (00 as input) or 1 for a wire (input 1)).
- * @return sidb_energy_and_state_type Electrostatic potential energy of all charge distributions with state type.
+ * @param valid_charge_distributions Physically valid charge distributions.
+ * @param output_bdl_pairs Output BDL pairs.
```

Comment:
Why are these not determined automatically via `detect_bdl_pairs`?

Reply:
@marcelwa The reason for this was to save runtime because this function can be called very often while ``output_bdl_pairs`` remains the same.

Reply:
I understand